ssh/validate.py

```diff
@@ -63,30 +66,55 @@ def check_ipv4_addrs(ips):
             socket.inet_pton(socket.AF_INET, str(ip))
         except OSError:
             invalid_ips.append(ip)
-    assert not len(invalid_ips), ('Only IPv4 values are allowed. The following are invalid IPv4 addresses: '
-                                  '{}'.format(invalid_ips))
+    assert not invalid_ips, ('Only IPv4 values are allowed. The following are invalid IPv4 addresses: '
+                             '{}'.format(invalid_ips))
 
 
-def validate_config(config):
-    assert isinstance(config, dict)
+def validate_optional_agent(agent_list, public_agent_list):
+    if agent_list:
+        validate_hosts_list(agent_list, 'agent_list')
 
-    ssh_keys_checks_map = {
-        'ssh_user': validate_ssh_user,
-        'ssh_port': validate_ssh_port,
-        'ssh_key_path': validate_ssh_key_path,
-        'agent_list': validate_hosts_list,
-        'master_list': lambda master_list: validate_master_agent_lists(master_list, config.get('agent_list'))
-    }
+    if public_agent_list:
+        validate_hosts_list(public_agent_list, 'public_agent_list')
+
+    if agent_list and public_agent_list:
+        compare_lists(agent_list, public_agent_list)
 
+
+def run_validate_config(value_func_map, config, required=True):
```

Comment:
`dev validate_config_chunk(key_validate_fn_map, config, keys_required)`
 - Make the function name closer to what the function is doing
 - Always require setting required. We don't call this from many places, defaults should be avoided